workflow/airflow_libero/dags/make_assets_public_dag.py

```diff
@@ -0,0 +1,238 @@
+    """
+
+    article_content = kwargs['ti'].xcom_pull(task_ids=None, key='article_content')
+
+    if article_content:
+        asset_uris = []
+
+        for content_item in article_content['content_items']:
+            xml = BeautifulSoup(content_item['content'], 'lxml-xml')
+            content_uris = []
+
+            for ele in ASSET_ELEMENTS:
+                for var in xml.find_all(ele):
+                    if var.attrs['media-type'] in SUPPORTED_MEDIA_TYPES:
+                        content_uris.append(var.contents[0])
+
+            asset_uris += content_uris
+
+        kwargs['ti'].xcom_push('asset_uris', json.dumps({'asset_uris': asset_uris}))
+        return SUCCESS
+    return FAILURE
+
+
+def download_assets(*args, **kwargs) -> bool:
+    """Download some static assets.
+
+    :return: bool
+    """
+    article_id = kwargs['ti'].xcom_pull(task_ids=None, key='article_id')
+    asset_uris = kwargs['ti'].xcom_pull(task_ids=None, key='asset_uris')
+
+    data_dir = os.path.join(PUBLIC_DATA_DIR, article_id)
+    if not os.path.isdir(data_dir):
+        os.makedirs(data_dir)
+
+    if asset_uris:
+        uris = json.loads(asset_uris.replace("'", '"'))
+
+        for uri in uris['asset_uris']:
+            response = requests.get(url=uri)
+            file_name = uri.split('/')[-1]
+
+            with open(os.path.join(data_dir, file_name), 'wb') as asset_file:
+                asset_file.write(response.content)
+
+
+def update_asset_uris(*args, **kwargs) -> bool:
+    """Update asset URIs in XML article content.
+
+    :return: bool
+    """
+    article_id = kwargs['ti'].xcom_pull(task_ids=None, key='article_id')
+    article_content = kwargs['ti'].xcom_pull(task_ids=None, key='article_content')
+
+    if article_content:
+        for content_item in article_content['content_items']:
+            xml = BeautifulSoup(content_item['content'], 'lxml-xml')
+
+            for ele in ASSET_ELEMENTS:
+                for var in xml.find_all(ele):
+                    if var.attrs['media-type'] in SUPPORTED_MEDIA_TYPES:
+                        file_name = var.contents[0].split('/')[-1]
+                        var.string = f'{STATIC_ASSETS_URL}/articles/{article_id}/{file_name}'
+
+            content_item['content'] = str(xml)
+
+        kwargs['ti'].xcom_push('article_content', article_content)
+        return SUCCESS
+    return FAILURE
+
+
+def deposit_to_article_store(*args, **kwargs) -> bool:
+    run_id = kwargs['ti'].xcom_pull(task_ids=None, key='run_id')
+    article_id = kwargs['ti'].xcom_pull(task_ids=None, key='article_id')
+    article_version = kwargs['ti'].xcom_pull(task_ids=None, key='article_version')
+    article_content = kwargs['ti'].xcom_pull(task_ids=None, key='article_content')
+
+    url = f'http://article-store:8000/articles/{article_id}/versions/{article_version}'
+    payload = convert_to_xml(article_content)
+
+    headers = {
+        'X-LIBERO-RUN-ID': run_id,
+        'X-LIBERO-AIRFLOW': 'true',
```

Comment:
I hadn't thought about using the public API for this communication, simpler than having another mechanism, but would probably need something more to limit Airflow bypassing to Airflow.